forms/TagList/index.js

```diff
@@ -8,38 +8,28 @@ module.exports = React.createClass({
   displayName: 'TagList',
 
   propTypes: {
-    onItemRemoved: React.PropTypes.func,
+    onItemIconClicked: React.PropTypes.func,
```

Comment:
Do you think this could be called `onItemClick`?

I'm also thinking we could call the `items` `tags` instead.